File: backend/src/main/java/isep/webapp/terasync/model/RatingList.java
package isep.webapp.terasync.model;

import jakarta.persistence.Entity;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class RatingList extends isep.webapp.terasync.model.Entity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ratingListId")
    private int ratingListId;
    @Column(name = "rating")
    private double rating;
    @Column(name = "comment")
    private String comment;
    @CreationTimestamp
    @Column(name = "ratingDate")
    private String ratingDate;
    @Column(name = "customerId")
    private int customerId;
    @Column(name = "businessId")
    private int businessId;
    @Column(name = "customerOrderId")
    private int customerOrderId;

//    @ManyToOne(
//            fetch = FetchType.EAGER,
//            targetEntity = CustomerOrder.class
//    )
//    @JoinColumn(
//            name = "customerOrderId",
//            insertable = false,
//            updatable = false
//    )
//    private CustomerOrder customerOrder;
}

File: front/src/app/model/rating.list.ts
export class RatingList {
  ratingListId: number | undefined;
  rating: number;
  comment: string;
  ratingDate: string | undefined;
  customerId: number;
  businessId: number;
  customerOrderId: number;


  constructor(rating: number, comment: string, customerId: number, businessId: number, customerOrderId: number,
              ratingDate?: string, ratingListId?: number) {
    this.ratingListId = ratingListId;
    this.rating = rating;
    this.comment = comment;
    this.ratingDate = ratingDate;
    this.customerId = customerId;
    this.businessId = businessId;
    this.customerOrderId = customerOrderId;
  }

  public static fromJson(json: RatingList): RatingList {
    return new RatingList(json.rating, json.comment, json.customerId, json.businessId, json.customerOrderId,
      json.ratingDate, json.ratingListId);
  }

  static initializeRatingLists(jsonUser: {ratingLists: RatingList[] | undefined}): RatingList[] {
    let ratingLists: RatingList[] = [];
    if (jsonUser.ratingLists != undefined) {
      for (let ratingList of jsonUser.ratingLists) {
        ratingLists.push(RatingList.fromJson(ratingList));
      }
    }
    return ratingLists;
  }
}

